### DISCUSSION TITLE
Auto-inbetweening

### DISCUSSION THREAD

**Discussion by andydjames** [keiwando/toonsquid#609] [Ideas]
My dream new feature that would knock any other feature out of the park would be vector interpolation (or auto-inbetweening) _that supports onion skinning_.
Cacani is the king of this feature but OpenToonz does a pretty good job too without bells and whistles that Cacani can offer (neither available on iPad).

**Already possible in ToonSquid**
Using the Path Tool, you can shape-morph shapes and lines (with the Freehand tool) over however many frames by keyframing control points.
The problem is that it kind of gets messy keeping track of all the keyframes, synching keyframes between different shapes and doesn't support onion skinning. As a result it can easily lose the hand-drawn feel.

**Enhanced feature suggestion 1**
Building on the existing shape tweening, provide the option to "Split into Drawings". So vector shape/line morphs over many frames can be split into ones or twos so you then have a series of individual editable vector drawings. With onion skinning you can further manipulate these drawings.
Currently, if you split a shape tween, the keyframes are reset and the control point keyframes are pushed so everything is out of synch.

**Enhanced feature suggestion 2 (the better way for animating)**
The other way of doing the same thing that would properly support a more hand-drawn approach is as follows:

- Set drawing one as a keyframe 1 on your timeline, draw your vector shape or freehand path (line). 
- Set drawing two as keyframe 2 further along your timeline, duplicate or copy keyframe 1 and manipulate into your 2nd keyframe pose.
- Set the intervening frames as "inbetween" frames.
- Apply interpolation/auto-inbetweening: The vector shape/line interpolates between keyframe 1 and 2 and populates the inbetween frames on ones or twos based on your inbetween frame drawings. 
- Edit your keyframes, apply easing, add/subtract inbetween frames, and regenerate the auto-inbetweens as many times as you need to.

This means you can see the individual drawings, move the individual drawings, track the relative movement with onion skinning. 
Your animation workflow then is to create your key poses, auto-inbetween, edit. It takes practice but is a massive time saver and you have all the benefits of vector line work.

**Comment by theanchhay** [comment]
This would be such a powerful feature to have. Toonsquid is currently one of the best animation app already, but this feature would elevate it so much. I've not used Cacani, but they got this smart painting system that also looks like an amazing feature to have.
